### PR TITLE
#4538: fix context menu error display; add ability to disable context menu success message

### LINKED ISSUE #4538: Context Menu error alert shows blueprint JSON in error alert to end-user
Context
---
- Seen on 1.7.11

Steps to Reproduce
---
- Write a context menu that uses an undefined variable

Screenshot
---
image

image

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #4538 
- Fixes context menu to show a generic error message instead of JSON blob of error
- Adds affordance to turn off the context menu success message. (Success message is on by default)
- Drops some dead code from the Page Editor adapter definitions (we don't support using a pre-made foundation). Removes the `GenericInsertPane` file in favor of a `useAutoInsert` hook for auto inserting

## Demo

- https://www.loom.com/share/9a215236cb1f42c2a0a0c4f597cf8f2d

## Future Work

- Consider trying to select error message detail to show in the error toast. Currently the user has to look in the logs to see what the real error was. Applies to button and other extensions, too

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer: @turbochef

### CODE REVIEW

File: src/background/contextMenus.ts [hunk 1]
```diff
@@ -85,22 +83,13 @@ async function dispatchMenu(
...
-      // especially deep inside other objects.
-      reportError(error);
-
-      notify.error(target, {
-        message: "Error handling context menu action",
-        error,
-        reportError: false,
-      });
-    }
+    // Handle internal/messenger errors here. The real error handling occurs in the contextMenu extension point
```
Comment (twschiller) [inline src/background/contextMenus.ts:87]: Moved the primary message logic to the extension point definition file

File: src/pageEditor/extensionPoints/contextMenu.test.ts [hunk 2]
```diff
@@ -15,18 +15,24 @@
...
  */
 
-.buttonRow {
-  padding-left: 6rem;
-  padding-right: 6rem;
-  display: flex;
-  flex-direction: column;
-  align-items: stretch;
-}
+import config from "@/pageEditor/extensionPoints/contextMenu";
```
Comment (twschiller) [inline src/pageEditor/extensionPoints/contextMenu.test.ts:18]: I dropped a file, and Git got confused

File: src/pageEditor/extensionPoints/contextMenu.ts [hunk 3]
```diff
@@ -184,19 +184,6 @@ const config: ElementConfig<undefined, ContextMenuFormState> = {
   selectExtensionPointConfig,
   selectExtension,
   fromExtension,
-  InsertModeHelpText: () => (
```
Comment (twschiller) [inline src/pageEditor/extensionPoints/contextMenu.ts:187]: This messaging is the dead code we dropped from all the extension point types

File: src/pageEditor/extensionPoints/quickBar.ts [hunk 4]
```diff
@@ -177,69 +169,6 @@ function asDynamicElement(element: QuickBarFormState): DynamicDefinition {
   };
 }
 
-export const UnconfiguredQuickBarAlert: React.FunctionComponent = () => {
```
Comment (twschiller) [inline src/pageEditor/extensionPoints/quickBar.ts:180]: Moved to the only file it's used in now

File: src/pageEditor/panes/insert/useAutoInsert.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  updateDynamicElement,
+} from "@/contentScript/messenger/api";
+import { reportEvent } from "@/telemetry/events";
+import { type ExtensionPointType } from "@/extensionPoints/types";
+import { ADAPTERS } from "@/pageEditor/extensionPoints/adapter";
+import notify from "@/utils/notify";
+
+const { addElement, toggleInsert } = actions;
+
+export function useAutoInsert(type: ExtensionPointType): void {
```
Comment (twschiller) [inline src/pageEditor/panes/insert/useAutoInsert.ts:19]: Moved the business logic from GenericInsert pane to this hook

File: src/pageEditor/tabs/contextMenu/ContextMenuConfiguration.tsx [hunk 6]
```diff
@@ -45,90 +47,106 @@ export const contextOptions = [
...
-      description={
-        <span>
-          The context menu item caption. Use the <code>%s</code> placeholder to
-          have the browser dynamically insert the current selection in the menu
-          caption
-        </span>
-      }
-    />
+}> = ({ isLocked = false }) => {
+  const [{ value: onSuccess }] = useField("extension.onSuccess");
```
Comment (twschiller) [inline src/pageEditor/tabs/contextMenu/ContextMenuConfiguration.tsx:51]: View this one with whitespace disabled. Raised the nesting level due to introducing the hook. This PR just adds 1 more field

File: src/pageEditor/tabs/editTab/FoundationNodeConfigPanel.tsx [hunk 7]
```diff
@@ -23,10 +23,38 @@ import useFlags from "@/hooks/useFlags";
...
 import { ADAPTERS } from "@/pageEditor/extensionPoints/adapter";
 import { useSelector } from "react-redux";
 import { selectActiveElement } from "@/pageEditor/slices/editorSelectors";
+import useQuickbarShortcut from "@/hooks/useQuickbarShortcut";
+import { Alert } from "react-bootstrap";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faExclamationTriangle } from "@fortawesome/free-solid-svg-icons";
+import { openShortcutsTab, SHORTCUTS_URL } from "@/chrome";
+
+const UnconfiguredQuickBarAlert: React.FunctionComponent = () => {
```
Comment (twschiller) [inline src/pageEditor/tabs/editTab/FoundationNodeConfigPanel.tsx:35]: Moved from the quick bar adapter file

File: src/pageEditor/tabs/contextMenu/ContextMenuConfiguration.tsx [hunk 8]
```diff
@@ -45,90 +47,106 @@ export const contextOptions = [
...
+            </p>
+          }
+          {...makeLockableFieldProps("Target Mode", isLocked)}
+        >
+          <option value="eventTarget">eventTarget</option>
+          <option value="document">document</option>
+          <option value="legacy">legacy</option>
+        </ConnectedFieldTemplate>
+
+        {(typeof onSuccess === "boolean" || onSuccess == null) && (
```
Comment (twschiller) [inline src/pageEditor/tabs/contextMenu/ContextMenuConfiguration.tsx:121]: This is the new field

**Review by BLoe — APPROVED** [review]

**Pull request merged**